newhelm/main.py

```diff
@@ -60,6 +61,37 @@ def list_tests() -> None:
         click.echo()
 
 
+@newhelm_cli.command()
+def list_suts():
+    """List details about all registered SUTs (System Under Test)."""
+    secrets = load_secrets_from_config()
+
+    for sut, sut_entry in SUTS.items():
+        try:
+            _, _, used_secrets = inject_dependencies(sut_entry.args, sut_entry.kwargs, secrets)
+            
+            flag = "Uses"
```

Comment:
Ideally if a SUT had one secret provide and one missing, we could report both of those at the same time.

Reply:
No problem! I'm glad to contribute code improvements. :) I've restored the `inject_dependencies` function and introduced `list_dependency_usage`. Currently, it doesn't return optional secrets, but but I will add it when I have time soon. Also, I've updated the formatting for better clarity: now, if there are both used and missing requirements, they will be displayed simultaneously. If there are no missing requirements, they won't be shown. Additionally, in cases where no secrets are needed, the message No Secrets Used. will be displayed.

Example for sut without used secrets and with missing secrets:
```shell
$ poetry run python newhelm/main.py list-suts

llama-2-7b
Class: TogetherCompletionsSUT
Args: ('togethercomputer/llama-2-7b', <newhelm.secret_values.InjectSecret object at 0x00000228F06DBCD0>)
Kwargs: {}
No Secrets Used.
Missing Secrets:
Scope: 'together', Key: 'api_key', Instructions: 'See https://api.together.xyz/settings/api-keys'
```

Example for sut with used secrets and without missing secrets:
```shell
$ poetry run python newhelm/main.py list-suts

gpt2
Class: HuggingFaceSUT
Args: ('gpt2', <newhelm.secret_values.InjectSecret object at 0x0000022886B9BA30>)
Kwargs: {}
Used Secrets:
[HuggingFaceToken()]
```

Example for sut without both used and missing secrets:
```shell
$ poetry run python newhelm/main.py list-suts

Class: DemoYesNoSUT
Args: ()
Kwargs: {}
No Secrets Used.
```

Example for sut with both used and missing secrets:
```shell
$ poetry run python newhelm/main.py list-suts

gpt-3.5-turbo
Class: OpenAIChat
Args: ('gpt-3.5-turbo', <newhelm.secret_values.InjectSecret object at 0x00000228F0D7A110>, <newhelm.secret_values.InjectSecret object at 0x00000228F0D7AC80>)
Kwargs: {}
Used Secrets:
[OpenAIOrgId()]
Missing Secrets:
Scope: 'openai', Key: 'api_key', Instructions: 'See https://platform.openai.com/api-keys'
```

I'm sorry that I accidentally used the name of the internal function in the commit message because I was distracted.

Reply:
@brianwgoldman  No problem! I'm glad to contribute code improvements. :) I've restored the `inject_dependencies` function and introduced `list_dependency_usage`. Currently, it doesn't return optional secrets, but but I will add it when I have time soon. Also, I've updated the formatting for better clarity: now, if there are both used and missing requirements, they will be displayed simultaneously. If there are no missing requirements, they won't be shown. Additionally, in cases where no secrets are needed, the message No Secrets Used. will be displayed.

Example for sut without used secrets and with missing secrets:
```shell
$ poetry run python newhelm/main.py list-suts

llama-2-7b
Class: TogetherCompletionsSUT
Args: ('togethercomputer/llama-2-7b', <newhelm.secret_values.InjectSecret object at 0x00000228F06DBCD0>)
Kwargs: {}
No Secrets Used.
Missing Secrets:
Scope: 'together', Key: 'api_key', Instructions: 'See https://api.together.xyz/settings/api-keys'
```

Example for sut with used secrets and without missing secrets:
```shell
$ poetry run python newhelm/main.py list-suts

gpt2
Class: HuggingFaceSUT
Args: ('gpt2', <newhelm.secret_values.InjectSecret object at 0x0000022886B9BA30>)
Kwargs: {}
Used Secrets:
[HuggingFaceToken()]
```

Example for sut without both used and missing secrets:
```shell
$ poetry run python newhelm/main.py list-suts

Class: DemoYesNoSUT
Args: ()
Kwargs: {}
No Secrets Used.
```

Example for sut with both used and missing secrets:
```shell
$ poetry run python newhelm/main.py list-suts

gpt-3.5-turbo
Class: OpenAIChat
Args: ('gpt-3.5-turbo', <newhelm.secret_values.InjectSecret object at 0x00000228F0D7A110>, <newhelm.secret_values.InjectSecret object at 0x00000228F0D7AC80>)
Kwargs: {}
Used Secrets:
[OpenAIOrgId()]
Missing Secrets:
Scope: 'openai', Key: 'api_key', Instructions: 'See https://platform.openai.com/api-keys'
```

I'm sorry that I accidentally used the name of the internal function in the commit message because I was distracted.